src/main/java/Board.java

```diff
@@ -39,29 +39,35 @@ public void clearBoard() {
         }
     }
 
+    /**
+     * Cells are considered adjacent in all directions, even diagonally
+     * @param cellPosition find cells adjacent to the given cellPosition
+     * @return a Set containing all the cells adjacent to the given position
+     * @throws InvalidPositionException when the given position is invalid
+     */
     @NotNull
-    public List<Cell> getAdjacentCells(@NotNull Position cellPosition) throws InvalidPositionException {
+    public Set<Cell> getAdjacentCells(@NotNull Position cellPosition) throws InvalidPositionException {
         if (cellPosition.getRow() > numberOfRows || cellPosition.getColumn() > numberOfColumns) {
             throw new InvalidPositionException("The specified position is outside the board");
         }
-        List<Cell> adjacentCells = new ArrayList<>();
+        Set<Cell> adjacentCells = new HashSet<>(4);
```

Comment:
Why 4? The maximum number of adjacent cells is 8

Reply:
You're right, I corrected it in a new commit